pkg/sfu/forwarder.go

```diff
@@ -1480,94 +1480,51 @@ func (f *Forwarder) getTranslationParamsVideo(extPkt *buffer.ExtPacket, layer in
 			f.rtpMunger.UpdateAndGetSnTs(extPkt) // call to update highest incoming sequence number and other internal structures
 			f.rtpMunger.PacketDropped(extPkt)
 			return tp, nil
-		} else if f.targetLayers.Spatial != f.currentLayers.Spatial && f.targetLayers.Spatial == layer && (extPkt.KeyFrame || tp.isSwitchingToTargetLayer) {
-			// lock to target layer
-			f.logger.Infow(
-				"locking to target layer",
-				"current", f.currentLayers,
-				"target", f.targetLayers,
-				"req", f.requestLayerSpatial,
-				"feed", extPkt.Packet.SSRC,
-			)
-			f.currentLayers.Spatial = f.targetLayers.Spatial
-			if !f.isTemporalSupported {
-				f.currentLayers.Temporal = f.targetLayers.Temporal
-			}
-			// TODO : we switch to target layer immediately now since we assume all frame chain is integrity
-			//   if we have frame chain check, should switch only if target chain is not broken and decodable
-			// if f.ddLayerSelector != nil {
-			// 	f.ddLayerSelector.SelectLayer(f.currentLayers)
-			// }
-			if f.currentLayers.Spatial >= f.maxLayers.Spatial {
-				tp.isSwitchingToMaxLayer = true
-			}
 		}
-	} else {
```

Comment:
This is a messy diff, but effectively using the previous `else` path for dependency descriptor case also.